sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -598,6 +606,15 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
       IntervalUtils.castStringToYMInterval(s, it.startField, it.endField))
     case _: YearMonthIntervalType => buildCast[Int](_, s =>
       IntervalUtils.periodToMonths(IntervalUtils.monthsToPeriod(s), it.endField))
+    case x: IntegralType if it.startField == it.endField =>
+      b =>
+        val intValue = try {
+          x.exactNumeric.asInstanceOf[Numeric[Any]].toInt(b)
```

Comment:
how about we let `intToYearMonthInterval` take long? then only `intToYearMonthInterval` need to do overflow check, not here.

Reply:
make sense. It's a little weird `intToYearMonthInterval` take a parameter of long type. So I want to change the method name: `intToYearMonthInterval` => `integralToYearMonthInterval` and `longToDayTimeInterval` => `integralToDayTimeInterval`. Do you agree?

Reply:
+1